### ISSUE TITLE
Sometimes the on click event 'on click page' raises an error. 

### ISSUE THREAD

**Issue by WouterMansystems** [mendixlabs/charts#10]
Every now and then the on click page fails and results in a server side error. Can't figure out when, seems random. The widget used is a column chart with data source microflow. The entity from which the data is retrieved is non-persistent. The page that should be opened has a data view with this context set to the same non-persistent entity.

The full stack trace:
[Stack trace.txt](https://github.com/mendixlabs/charts/files/1689451/Stack.trace.txt)

**Comment by edwinmp** [comment]
Hi @WouterMansystems, I examined the stack trace you shared and I believe this is not an issue with the widget. The issue appears to originate from within the data source microflow. The widget only catches it. There seems to be an attempt to retrieve data from the database over an association to a non-persistent entity.
Anyway, the widget is built to catch errors that occur in the actions it calls, which is what's happening here.

Thanks.

**Comment by WouterMansystems** [comment]
Problem is, the widget fails to build the times the error occurs and the chart stays empty. This while, lets say, 4 out of 5 times there is no error and the chart is displayed.

**Comment by Andries-Smit** [comment]
@WouterMansystems The platform does not support non persistent entities for Data source Database. Only Microflows can.

@edwinmp the XML says <property key="dataEntity" type="entity" allowNonPersistableEntities="true">
Though we need might need to do a run time validation on this for for data source Database.

**Comment by WouterMansystems** [comment]
I know, the data source is a microflow.

**Comment by WouterMansystems** [comment]
I tried the on click event 'run microflow' and there an error occurs as well. It may be related so I decided to post it here. The second time I click on a bar to run the on click microflow the microflow misses the input parameter. See the attached video in the zip. There are no errors in the console, both server or client side. 

I'm using the latest version as available in the app store. 

[2018-04-26_13-28-49.zip](https://github.com/mendixlabs/charts/files/1950931/2018-04-26_13-28-49.zip)
